File: golang/_test/number_test.go
package random_test

import (
	"fmt"
	"github.com/stretchr/testify/require"
	"random/random"
	"testing"
)

func TestRandomNumber(t *testing.T) {
	tests := []struct {
		name string
		min  int
		max  int
	}{
		{
			name: "_test 1",
			min:  1,
			max:  5,
		},
		{
			name: "_test 2",
			min:  3,
			max:  5,
		},
		{
			name: "_test 3",
			min:  2,
			max:  10,
		},
		{
			name: "_test 4",
			min:  100,
			max:  200,
		},
	}
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			result := random.Int(test.min, test.max)
			require.Equal(t, true, result >= test.min && result <= test.max, fmt.Sprintf("result must be greater or euqal than %d and less or equal than %d", test.min, test.max))
		})
	}
}

File: rust/src/tests/number_test.rs
use crate::random::number::random_number;

struct TestNumberStruct {
    min: i32,
    max: i32,
}

#[test]
fn check_number() {
    let tests: [TestNumberStruct; 4] = [
        TestNumberStruct{ min: 1, max: 5 },
        TestNumberStruct{ min: 3, max: 5 },
        TestNumberStruct{ min: 2, max: 10 },
        TestNumberStruct{ min: 100, max: 200 },
    ];

    for test in tests {
        let result = random_number(test.min, test.max);
        println!("testing");
        assert!(result >= test.min || result <= test.max)
    }
}